futures/src/lib.rs

```diff
@@ -1,164 +1,28 @@
-//! Zero-cost Futures in Rust
+//! Abstractions for asynchronous programming
 //!
-//! This library is an implementation of futures in Rust which aims to provide
-//! a robust implementation of handling asynchronous computations, ergonomic
-//! composition and usage, and zero-cost abstractions over what would otherwise
-//! be written by hand.
+//! This crate provides a number of core abstractions for writing asynchronous code:
 //!
-//! Futures are a concept for an object which is a proxy for another value that
-//! may not be ready yet. For example issuing an HTTP request may return a
-//! future for the HTTP response, as it probably hasn't arrived yet. With an
-//! object representing a value that will eventually be available, futures allow
-//! for powerful composition of tasks through basic combinators that can perform
-//! operations like chaining computations, changing the types of futures, or
-//! waiting for two futures to complete at the same time.
+//! - [Futures](::Future) (sometimes called promises), which represent a single
+//! asychronous computation that may result in a final value or an error.
 //!
-//! You can find extensive tutorials and documentations at [https://tokio.rs]
-//! for both this crate (asynchronous programming in general) as well as the
-//! Tokio stack to perform async I/O with.
+//! - [Streams](::Stream), which represent a series of values or errors produced asynchronously.
 //!
-//! [https://tokio.rs]: https://tokio.rs
+//! - [Sinks](::Sink), which represent asynchronous writing of data.
```

Comment:
"represent asynchronous writing of data" doesn't sound quite right to me.
"which represent asynchonous pipes into which data can be written" is too wordy, but maybe something along those lines?